src/ks_json.c

```diff
@@ -451,6 +451,55 @@ KS_DECLARE(ks_json_t *) ks_json_enum_next(ks_json_t *item)
 	return item->next;
 }
 
+KS_DECLARE(ks_bool_t) ks_json_replace_item(ks_json_t *parent, ks_json_t **item, ks_json_t *replacement)
+{
+	if ((parent == NULL) || (replacement == NULL) || (item == NULL) || (*item == NULL)) {
+		return KS_FALSE;
+	}
+
+	replacement->next = (*item)->next;
+	replacement->prev = (*item)->prev;
+
+	if ((*item)->string) {
```

Comment:
Should it NULL the replacement if not set?

Reply:
@crienzo Any thoughts? I basically just copied this one over from your original code- nulling the replacement could have an impact in places calling the existing placeholder method (namely, resolving SWML variables I believe).

Reply:
let's leave it as-is for now

Reply:
I have a bit more memory now of what this does, if this helps. `string` is the name of the item. This is typically set (and required) when the parent is a JSON object. For arrays or standalone items, it's optional. The intended use-case for this method is to replace an item in an object (or array). Setting the string NULL will most likely break those use-cases unless we do an additional check on the parent ks_json_t object type.